src/psyclone/dynamo0p3.py

```diff
@@ -1215,10 +1215,16 @@ def __init__(self, kern_or_invoke):
             # field proxy and undf is not required.
             if self._invoke and self._invoke.field_on_space(function_space):
                 if not (self._dofs_only and Config.get().distributed_memory):
-                    self._var_list.append(function_space.undf_name)
+                    if not self._invoke.operates_on_dofs_only:
```

Comment:
How does `self._dofs_only` compare with `self._invoke.operates_on_dofs_only`? Also, I hope we can get rid of these changes.